### PR TITLE
Refactor isAuthenticatedUser method

### PR DESCRIPTION
**Pull request by DonOmalVindula**
### Purpose
Refactor isAuthenticatedUser method to accept strings as usernames of the current user and user to be checked to improve compatibility.

### Related Issues
- N/A

### Related PRs
- N/A

### Checklist
- [x] e2e cypress tests locally verified.
- [x] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation. - N/A
- [ ] Documentation provided. (Add links if there are any). - N/A
- [ ] Unit tests provided. (Add links if there are any). - N/A
- [ ] Integration tests provided. (Add links if there are any). - N/A

### Security checks
- [x] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?. - N/A
- [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

**Review by DimalChandrasiri — APPROVED** [review]

File: apps/console/src/features/users/pages/user-edit.tsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /**
- * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ * Copyright (c) 2022, WSO2 LLC. (https://www.wso2.com) All Rights Reserved.
```
Comment (NipuniBhagya) [inline apps/console/src/features/users/pages/user-edit.tsx:2]:
```suggestion
 * Copyright (c) 2020, WSO2 LLC. (https://www.wso2.com) All Rights Reserved.
```

File: apps/console/src/features/users/utils/user-management-utils.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 /**
- * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ * Copyright (c) 2022, WSO2 LLC. (https://www.wso2.com) All Rights Reserved.
```
Comment (NipuniBhagya) [inline apps/console/src/features/users/utils/user-management-utils.ts:2]:
```suggestion
 * Copyright (c) 2020, WSO2 LLC. (https://www.wso2.com) All Rights Reserved.
```

File: apps/console/src/features/users/utils/user-management-utils.ts [hunk 3]
```diff
@@ -28,21 +25,39 @@ export class UserManagementUtils {
...
-     * @param {ProfileInfoInterface} authenticatedUser - Current authenticated user.
-     * @param {ProfileInfoInterface} evaluatingUser - Profile of the evaluating user.
-     * @return {boolean} If the user is the authenticated user or not.
+     * @param authenticatedUser - Currently logged in user
+     * @param user - User object.
+     * @returns Whether the given user is the current user or not.
      */
-    public static isAuthenticatedUser(authenticatedUser: ProfileInfoInterface,
-                                      evaluatingUser: ProfileInfoInterface): boolean {
+    public static isAuthenticatedUser = (authenticatedUser: string, username: string): boolean => {
```
Comment (NipuniBhagya) [inline apps/console/src/features/users/utils/user-management-utils.ts:53]: Better to verify the authenticated user with the user ID

**Review by NipuniBhagya — CHANGES_REQUESTED** [review]

**Review by NipuniBhagya — APPROVED** [review]

File: apps/console/src/features/users/utils/user-management-utils.ts [hunk 4]
```diff
@@ -28,21 +25,39 @@ export class UserManagementUtils {
...
     // eslint-disable-next-line @typescript-eslint/no-empty-function
     private constructor() { }
 
+    /**
+     * Resolves the username of the given user.
+     *
+     * @param user - User object.
+     * @returns The username of the given user as a string.
+     */
+    public static resolveUsername = (username: string): string => {
```
Comment (NipuniBhagya) [inline apps/console/src/features/users/utils/user-management-utils.ts:38]: We can use this method instead https://github.com/wso2/identity-apps/blob/master/modules/core/src/helpers/profile.ts#L123

Comment (DonOmalVindula) [reply apps/console/src/features/users/utils/user-management-utils.ts]: Resolved with 8413b271b71daf184e1b29a2ed740b1b046cdfd9

File: apps/console/src/features/users/pages/user-edit.tsx (hunk 1 above)
Comment (DonOmalVindula) [reply apps/console/src/features/users/pages/user-edit.tsx]: Resolved with 8413b271b71daf184e1b29a2ed740b1b046cdfd9

File: apps/console/src/features/users/utils/user-management-utils.ts (hunk 3 above)
Comment (DonOmalVindula) [reply apps/console/src/features/users/utils/user-management-utils.ts:41]: Resolved with 8413b271b71daf184e1b29a2ed740b1b046cdfd9

**Pull request merged**